src/Sentry/Internal/Hub.cs

```diff
@@ -342,12 +342,27 @@ public void CaptureTransaction(Transaction transaction)
             {
                 // Apply scope data
                 var currentScope = ScopeManager.GetCurrent();
-                currentScope.Key.Evaluate();
-                currentScope.Key.Apply(transaction);
+                var scope = currentScope.Key;
+                scope.Evaluate();
+                scope.Apply(transaction);
 
                 // Apply enricher
                 _enricher.Apply(transaction);
 
+                if (transaction.IsSampled != false)
```

Comment:
I'm concerned that if we rely on the user setting `IsSampled = false` in a transaction processor to indicate that they want to discard the transaction, we might mix up our discard reasons for client reports.

If we used the nullable return value pattern like we do for event processors, we wouldn't have that concern.